### ISSUE TITLE
Plugin test suites

### ISSUE THREAD

**Issue by JGAntunes** [netlify/plugins#212]
Coming from a discussion that started in slack during a plugin review, even though right now we’re not [looking at styling and programming patterns](https://github.com/netlify/plugins/blob/master/docs/plugin_review.md#process-for-reviewers), however should we look into how these plugins are tested and if they have automated test suites? Should we  "enforce" this at some level?

I understand we’re aiming at having a low barrier of entry for contributors, but If we’re “vetting” these plugins to be part of our plugin list I wonder if we shouldn’t be more picky about testing its functional reqs?

**Comment by erezrokah** [comment]
Good point. It might make plugin reviews longer though - do we verify the correctness of automated tests?

We could start by running linting ourselves on plugins' source code and only use rules that clearly find bugs (and not styling issues). For example this one https://eslint.org/docs/rules/no-dupe-keys

**Comment by ingride** [comment]
I also think we should consider the security aspect of it? how will we validate the plugins are not doing something they shouldn't or sending / gathering data they shouldn't.
cc @dustincrogers

**Comment by ehmicky** [comment]
Linting would be great. 

That being said, in my opinion, most bugs in plugins are about some feature-related behavior that is not possible to determine programmatically, especially with only checking a JavaScript AST. However, security issues discovery could definitely be partially discovered using one of the many libraries doing this.

When it comes to automated tests, I feel like we should definitely encourage it. But I am concerned enforcing it would hinder community contributions too long and make plugin reviews more tedious for both us and the authors as @erezrokah is mentioning.
